### ISSUE TITLE
[BUG]: Invalid deployment action: Script. For SQL query operations, deployment actions should be set to value `Publish`

### ISSUE THREAD

**Issue by sgabler** [microsoft/azure-pipelines-tasks#18711]
### Task name

SqlAzureDacpacDeployment

### Task version

1

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [X] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Operation system

windows-2022

### Task log

```shell
2023-07-19T16:39:41.7730913Z ##[section]Starting: Configure AD Access for stage users
2023-07-19T16:39:41.8593677Z ==============================================================================
2023-07-19T16:39:41.8594362Z Task         : Azure SQL Database deployment
2023-07-19T16:39:41.8594768Z Description  : Deploy an Azure SQL Database using DACPAC or run scripts using SQLCMD
2023-07-19T16:39:41.8594931Z Version      : 1.225.1
2023-07-19T16:39:41.8595166Z Author       : Microsoft Corporation
2023-07-19T16:39:41.8595528Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/deploy/sql-azure-dacpac-deployment
2023-07-19T16:39:41.8595704Z ==============================================================================
2023-07-19T16:40:00.8690519Z ##[error]Invalid deployment action: Script. For SQL query operations, deployment actions should be set to value `Publish`. Check out how to troubleshoot failures at https://aka.ms/sqlazuredeployreadme#troubleshooting-
2023-07-19T16:40:00.8968015Z ##[section]Finishing: Configure AD Access for stage users
```

### Relevant log output

```shell
2023-07-19T16:40:00.8690519Z ##[error]Invalid deployment action: Script. For SQL query operations, deployment actions should be set to value `Publish`. Check out how to troubleshoot failures at https://aka.ms/sqlazuredeployreadme#troubleshooting-
```

### Aditional info

This bug was already reported here: https://github.com/microsoft/azure-pipelines-tasks/issues/10479, but unfortunately closed without a satisfying result.

[The offered workaround](https://github.com/microsoft/azure-pipelines-tasks/issues/10479#issuecomment-496073181) works, but it is highly unintuitive. My colleagues and me spent a total of 4 hours to find this solution. You can also see that the feedback in the original issue is unhappy with the offered solution.

I would kindly ask you to revisit this topic to avoid that other developers waste time here. Thank you.

In short: if I select the deploy type "SQL Script File" in the UI, then I should not be blocked by a hidden setting that is only visible if I select "SQL DACPAC File", which is completely unrelated to what I am trying to achieve.
